### PR TITLE
fix: fixed a case for accesing a private field

### PR DESCRIPTION
**Pull request by Bayheck**
fixed a case for accesing a private field

### CODE REVIEW

File: esotope.js [hunk 1]
```diff
@@ -1313,8 +1313,12 @@ var ExprRawGen = {
             _.js += ']';
         }
 
-        else
-            _.js += ($expr.optional ? '?.' : '.') + $prop.name;
+        else {
+            const propName     = $prop.type === Syntax.PrivateIdentifier ? "#" + $prop.name : $prop.name;
```
Comment (Aleksey28) [inline esotope.js]: It's preferable to use single quotes for strings

File: esotope.js [hunk 2]
```diff
@@ -1313,8 +1313,12 @@ var ExprRawGen = {
...
+            const exprChaining = $expr.optional ? '?.' : '.';
```
Comment (Aleksey28) [inline esotope.js]: There are a few problems with the name `exprChaining`:
1. For what word is the `expr` short form, `express` or `expression`? If the first one it means that it's verb and name `exprChaining` should be used for method or function. Try not to reduce names without reason and if it's not a common name
2. Try to look at the name `exprChaining` without context and answer the question `What does it contain?`. My first thought was that it was an expression sequence connected in one chain, but it's not.
My suggestion for the name of this variable is `chainOperator`

**Review by Aleksey28 — CHANGES_REQUESTED** [review]
That's great that you found out the root of the issue. I left some comments about the code style

File: esotope.js [hunk 3]
```diff
@@ -1313,8 +1313,12 @@ var ExprRawGen = {
             _.js += ']';
         }
 
-        else
-            _.js += ($expr.optional ? '?.' : '.') + $prop.name;
+        else {
+            const propName      = $prop.type === Syntax.PrivateIdentifier ? '#' + $prop.name : $prop.name;
```
Comment (AlexKamaev) [inline esotope.js]: there is very insignificant code duplication with $prop.name and $prop.name.
I would rewrite it as:
```JS
const prefix = $prop.type === Syntax.PrivateIdentifier ? '#' : '';

const propName = $prefix + $prop.name;
```

**Review by AlexKamaev — CHANGES_REQUESTED** [review]

**Review by Aleksey28 — APPROVED** [review]

**Review by AlexKamaev — APPROVED** [review]

**Pull request merged**